qiskit_experiments/calibration/calibrations.py

```diff
@@ -0,0 +1,972 @@
+        qubits = self._to_tuple(qubits)
+
+        # 1) Identify the parameter object.
+        param_name = param.name if isinstance(param, Parameter) else param
+        sched_name = schedule.name if isinstance(schedule, ScheduleBlock) else schedule
+
+        param = self.calibration_parameter(param_name, qubits, sched_name)
+
+        # 2) Get a list of candidate keys restricted to the qubits of interest.
+        candidate_keys = []
+        for key in self._parameter_map_r[param]:
+            candidate_keys.append(ParameterKey(key.parameter, qubits, key.schedule))
+
+        # 3) Loop though the candidate keys to candidate values
+        candidates = []
+        for key in candidate_keys:
+            if key in self._params:
+                candidates += self._params[key]
+
+        # If no candidate parameter values were found look for default parameters
+        # i.e. parameters that do not specify a qubit.
+        if len(candidates) == 0:
+            for key in candidate_keys:
+                if ParameterKey(key.parameter, (), key.schedule) in self._params:
+                    candidates += self._params[ParameterKey(key.parameter, (), key.schedule)]
+
+        # 4) Filter candidate parameter values.
+        if valid_only:
+            candidates = [val for val in candidates if val.valid]
+
+        candidates = [val for val in candidates if val.group == group]
+
+        if cutoff_date:
+            candidates = [val for val in candidates if val.date_time <= cutoff_date]
+
+        if len(candidates) == 0:
+            msg = f"No candidate parameter values for {param_name} in calibration group {group} "
+
+            if qubits:
+                msg += f"on qubits {qubits} "
+
+            if sched_name:
+                msg += f"in schedule {sched_name} "
+
+            if cutoff_date:
+                msg += f"with cutoff date: {cutoff_date}"
+
+            raise CalibrationError(msg)
+
+        # 5) Return the most recent parameter.
+        return max(candidates, key=lambda x: x.date_time).value
+
+    def get_schedule(
+        self,
+        name: str,
+        qubits: Union[int, Tuple[int, ...]],
+        assign_params: Dict[Union[str, ParameterKey], ParameterValueType] = None,
+        group: Optional[str] = "default",
+        cutoff_date: datetime = None,
+    ) -> ScheduleBlock:
+        """Get the schedule and assign values to parameters.
+
+        The parameters in the template schedule block will be assigned to the values managed
+        by the calibrations unless they are specified in assign_params. In this case the value in
+        assign_params will override the value stored by the calibrations. A parameter value in
+        assign_params may also be a :class:`ParameterExpression`.
+
+        Args:
+            name: The name of the schedule to get.
+            qubits: The qubits for which to get the schedule.
+            assign_params: The parameters to assign manually. Each parameter is specified by a
```

Comment:
Can you show some code-block how to use assign_params? Having example is more intuitive.

Reply:
Yes. I'll add an example.

Reply:
done here: ba2bc1b